server/src/main/java/org/candlepin/dto/api/v1/CertificateSerialDTO.java

```diff
@@ -0,0 +1,184 @@
+        return this.serial;
+    }
+
+    public CertificateSerialDTO setSerial(BigInteger serial) {
+        this.serial = serial;
+        return this;
+    }
+
+    public Date getExpiration() {
+        return this.expiration;
+    }
+
+    public CertificateSerialDTO setExpiration(Date expiration) {
+        this.expiration = expiration;
+        return this;
+    }
+
+    public Boolean isCollected() {
+        return this.collected;
+    }
+
+    public CertificateSerialDTO setCollected(Boolean collected) {
+        this.collected = collected;
+        return this;
+    }
+
+    public Boolean isRevoked() {
+        return this.revoked;
+    }
+
+    public CertificateSerialDTO setRevoked(Boolean revoked) {
+        this.revoked = revoked;
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        Date expiration = this.getExpiration();
+        String date = expiration != null ? String.format("%1$tF %1$tT%1$tz", expiration) : null;
+
+        return String.format(
+            "CertificateSerialDTO [id: %s, serial: %s, expiration: %s, collected: %s, revoked: %s]",
+            this.getId(), this.getSerial(), date, this.isCollected(), this.isRevoked());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof CertificateSerialDTO && super.equals(obj)) {
+            CertificateSerialDTO that = (CertificateSerialDTO) obj;
+
+            EqualsBuilder builder = new EqualsBuilder()
+                .append(this.getId(), that.getId())
+                .append(this.getSerial(), that.getSerial())
+                .append(this.getExpiration(), that.getExpiration())
+                .append(this.isCollected(), that.isCollected())
+                .append(this.isRevoked(), that.isRevoked());
+
+            return builder.isEquals();
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+            .append(super.hashCode())
+            .append(this.getId())
+            .append(this.getSerial())
+            .append(this.getExpiration())
+            .append(this.isCollected())
+            .append(this.isRevoked());
+
+        return builder.toHashCode();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object clone() {
+        CertificateSerialDTO copy = (CertificateSerialDTO) super.clone();
+
+        copy.id = this.getId();
+        copy.serial = this.getSerial();
+        copy.expiration = this.getExpiration();
+        copy.collected = this.isCollected();
+        copy.revoked = this.isRevoked();
+
+        return copy;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CertificateSerialDTO populate(CandlepinDTO<CertificateSerialDTO> source) {
```

Comment:
Github is having trouble showing me the commit for this comment so I can scroll down one line, but I'll just assume it was something I fixed already. Current version looks like this:

```
    @Override
    public CertificateSerialDTO populate(CandlepinDTO<CertificateSerialDTO> source) {
        super.populate(source);
```